src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -285,6 +223,8 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleInfoPanelUpdate(infoPanelType);
             }
 
+            addNewUserInputToHistory(commandText);
```

Comment:
```suggestion
```

Reply:
should we shift this up to the top so that users can retrieve invalid commands too?

because I think it would be useful if the user typed a long command (eg some `addlesson` command) but made a typo; they can just press up arrow to retrieve it and edit it as necessary

Reply:
can shift it up to 
```java
 private CommandResult executeCommand(String commandText) throws CommandException, ParseException {

        // ✏️ shift up to here
        addNewUserInputToHistory(commandText);

        try {
            CommandResult commandResult = logic.execute(commandText);
            // ...
```

Reply:
I thought that currently if they input invalid command, it does not clear the command box?